### PR TITLE
feat: add sha512sum

### PR DESCRIPTION
**Pull request by mengzhuo**

### CODE REVIEW

File: cmds/core/shasum/shasum.go [hunk 1]
```diff
@@ -20,18 +21,21 @@ var algorithm int
...
 	switch algo {
 	case 1:
 		h = sha1.New()
 	case 256:
 		h = sha256.New()
+	case 512:
+		h = sha512.New()
 	default:
-		return nil, fmt.Errorf("invalid algorithm, only 1 or 256 are valid")
+		return nil, fmt.Errorf("invalid algorithm, only 1, 256 or 512 are valid")
```
Comment (rminnich) [inline cmds/core/shasum/shasum.go:38]: This patch could improve the basic code if you
fmt.Errorf("invalid algorithm, only 1, 256 or 512 are valid:%w", os.ErrInvalid)

Then you could change the test as below

File: cmds/core/shasum/shasum.go [hunk 2]
```diff
@@ -65,8 +69,8 @@ func shasum(w io.Writer, r io.Reader, args ...string) error {
 }
 
 func main() {
-	flag.IntVar(&algorithm, "algorithm", 1, "SHA algorithm, valid args are 1 and 256")
-	flag.IntVar(&algorithm, "a", 1, "SHA algorithm, valid args are 1 and 256 (shorthand)")
+	flag.IntVar(&algorithm, "algorithm", 1, "SHA algorithm, valid args are 1, 256 and 512")
```
Comment (rminnich) [inline cmds/core/shasum/shasum.go:72]: if you can take out the " (shorthand)" part of this text, that would be nice.

File: cmds/core/shasum/shasum_test.go [hunk 3]
```diff
@@ -50,6 +50,12 @@ func TestSHASum(t *testing.T) {
 			algorithm: 256,
 			want:      "ae0666f161fed1a5dde998bbd0e140550d2da0db27db1d0e31e370f2bd366a57 -\n",
 		},
+		{
```
Comment (rminnich) [inline cmds/core/shasum/shasum_test.go:53]: if you wrap os.ErrInvalid as suggested above, you can use errors.Is here.

The you can add
err error
to the struct, then initialize it with error values, then on line 108 you can use errors.Is. Just an idea.

**Review by rminnich — APPROVED** [review]
This code is an improvement over existing code, and my suggestions are not related to your PR , so feel free to ignore my suggestions. You or I or someone can improve it later. But if you want to implement my suggestions, thank you :-)

**Review by rminnich — APPROVED** [review]

**Comment by rminnich** [comment]
because this code follows the pattern of the original, I'm fine with submitting as is, lmk.

File: cmds/core/shasum/shasum_test.go (hunk 3 above)
Comment (mengzhuo) [reply cmds/core/shasum/shasum_test.go:55]: Done, PTAL

**Review by rminnich — APPROVED** [review]

**Pull request merged**